<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from bigman.urdf.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="bigman" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- 
  ROS urdf definition of the BIGMAN humanoid robot by Istituto Italiano di Tecnologia
  
  DO NOT INCLUDE THIS FILE DIRECTLY! USE bigman_robot.urdf.xacro INSTEAD!

  Author: Enrico Mingo & Alessio Rocchi

  Inertial matrix has to be in CoM frame (URDF specs)

  Velocity and effort limits are fake!
-->
  <!-- ************ MODEL CONFIGURATION ************ -->
  <!-- ***********  MODEL COLORS *********** -->
  <material name="grey">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <material name="dark_grey">
    <color rgba="0.3 0.3 0.3 1"/>
  </material>
  <material name="blue">
    <color rgba="0.0 0.2 0.3 1"/>
  </material>
  <!-- ************ BIGMAN BASE ************** -->
  <link name="base_link"/>
  <link name="Waist">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="-0.032862932 -0.00050930727 -0.0012899617"/>
      <mass value="6.4080904"/>
      <inertia ixx="0.038826078" ixy="0.00029047605" ixz="0.00068975598" iyy="0.025050374" iyz="-1.3370623e-05" izz="0.048055286"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="Waist_visual">
        <mesh filename="package://bigman_urdf/meshes/Waist.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="Waist_collision">
        <mesh filename="package://bigman_urdf/meshes/simple/Waist.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="base_joint" type="fixed">
    <parent link="base_link"/>
    <child link="Waist"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="RHipMot">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.10348188 -0.10919813 -0.064924056"/>
      <mass value="3.7958197"/>
      <inertia ixx="0.015639017" ixy="0.0018504661" ixz="0.0023907793" iyy="0.023519749" iyz="-0.0023303162" izz="0.020953643"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RHipMot_visual">
        <mesh filename="package://bigman_urdf/meshes/HipRoll.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="dark_grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RHipMot_collision">
        <mesh filename="package://bigman_urdf/meshes/simple/HipRoll.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="RThighUpLeg">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0052929709 -0.022561833 0.0056701669"/>
      <mass value="2.2402068"/>
      <inertia ixx="0.010102097" ixy="0.00010858487" ixz="0.00025291395" iyy="0.0067147" iyz="8.3447072e-05" izz="0.01048513"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RThighUpLeg_visual">
        <mesh filename="package://bigman_urdf/meshes/HipYaw.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="dark_grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RThighUpLeg_collision">
        <mesh filename="package://bigman_urdf/meshes/simple/HipYaw.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="RThighLowLeg">
    <inertial>
      <origin rpy="0 0 0" xyz="0.037746286 -0.028539939 -0.19310133"/>
      <mass value="5.0950736"/>
      <inertia ixx="0.051607858" ixy="0.0022253472" ixz="0.002452979" iyy="0.044349153" iyz="-0.0041006942" izz="0.023374453"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RThighLowLeg_visual">
        <mesh filename="package://bigman_urdf/meshes/HipPitch.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RThighLowLeg_collision">
        <mesh filename="package://bigman_urdf/meshes/simple/HipPitch.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="RLowLeg">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0050828668 0.0037732093 -0.072605017"/>
      <mass value="5.6495309"/>
      <inertia ixx="0.10535542" ixy="-0.00058199221" ixz="0.0016743598" iyy="0.097857796" iyz="-0.0033770181" izz="0.020854086"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RLowLeg_visual">
        <mesh filename="package://bigman_urdf/meshes/KneePitchCalf.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="dark_grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RLowLeg_collision">
        <mesh filename="package://bigman_urdf/meshes/simple/KneePitchCalf.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="RFootmot">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.022024173 0.0025677238 0.00094588455"/>
      <mass value="2.2844159"/>
      <inertia ixx="0.0075550679" ixy="0.00067488292" ixz="5.8332889e-05" iyy="0.011241163" iyz="-0.00014552081" izz="0.013391865"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RFootmot_visual">
        <mesh filename="package://bigman_urdf/meshes/AnklePitch.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="dark_grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RFootmot_collision">
        <mesh filename="package://bigman_urdf/meshes/simple/AnklePitch.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="RFoot">
    <inertial>
      <origin rpy="0 0 0" xyz="0.01282904 -0.00463694 -0.10109499"/>
      <mass value="3.83109981"/>
      <inertia ixx="0.01006627" ixy="-0.00051955" ixz="-0.00374069" iyy="0.02882577" iyz="0.00044505" izz="0.02819025"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry name="RFoot_visual">
        <mesh filename="package://bigman_urdf/meshes/Foot.STL" scale="0.001 -0.001 0.001"/>
      </geometry>
      <material name="dark_grey"/>
    </visual>
    <collision>
      <!--<origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry name="RFoot_collision">
        <mesh filename="package://bigman_urdf/meshes/simple/Foot.STL" scale="0.001 -0.001 0.001"/>
      </geometry>-->
      <origin rpy="0 0 0" xyz="0.047 -0.01 -0.1359"/>
      <geometry>
        <box size="0.306 0.16 0.015"/>
      </geometry>
    </collision>
  </link>
  <joint name="RHipLat" type="revolute">
    <parent link="Waist"/>
    <child link="RHipMot"/>
    <origin rpy="0 0 0" xyz="0.1059 -0.06 0.0"/>
    <axis xyz="1 0 0"/>
    <limit effort="220" lower="-0.87222" upper="0.697778" velocity="6.283"/>
    <dynamics damping="3.0" friction="0"/>
  </joint>
  <joint name="RHipYaw" type="revolute">
    <parent link="RHipMot"/>
    <child link="RThighUpLeg"/>
    <origin rpy="0 0 0" xyz="-0.0889 -0.121032 -0.217872"/>
    <axis xyz="0 0 1"/>
    <limit effort="120" lower="-1.57" upper="0.872222" velocity="6.283"/>
    <dynamics damping="3.0" friction="0"/>
  </joint>
  <joint name="RHipSag" type="revolute">
    <parent link="RThighUpLeg"/>
    <child link="RThighLowLeg"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="220" lower="-2.09333" upper="1.046667" velocity="6.283"/>
    <dynamics damping="3.0" friction="0"/>
  </joint>
  <joint name="RKneeSag" type="revolute">
    <parent link="RThighLowLeg"/>
    <child link="RLowLeg"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 -0.356"/>
    <axis xyz="0 1 0"/>
    <limit effort="220" lower="0.0" upper="2.442222" velocity="6.283"/>
    <dynamics damping="3.0" friction="0"/>
  </joint>
  <joint name="RAnkSag" type="revolute">
    <parent link="RLowLeg"/>
    <child link="RFootmot"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 -0.4"/>
    <axis xyz="0 1 0"/>
    <limit effort="220" lower="-1.39556" upper="0.697778" velocity="6.283"/>
    <dynamics damping="3.0" friction="0"/>
  </joint>
  <joint name="RAnkLat" type="revolute">
    <parent link="RFootmot"/>
    <child link="RFoot"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
    <axis xyz="1 0 0"/>
    <limit effort="120" lower="-0.785" upper="0.785" velocity="6.283"/>
    <dynamics damping="3.0" friction="0"/>
  </joint>
  <link name="r_leg_ft"/>
  <joint name="r_leg_ft_joint" type="fixed">
    <parent link="RFoot"/>
    <child link="r_leg_ft"/>
    <origin rpy="0 0 0" xyz="0.005 0 -0.107"/>
  </joint>
</robot>

